docs/docs/schema-customization.md

```diff
@@ -0,0 +1,765 @@
+    },
+  }
+  const resolvers = {
+    Query: {
+      allTeamMembers: {
+        type: ["TeamMember"],
+        resolve(source, args, context, info) {
+          return context.nodeModel.getAllNodes({ type: "TeamMember" })
+        },
+      },
+    },
+    AuthorJson: {
+      fullName,
+    },
+    ContributorJson: {
+      fullName,
+    },
+  }
+  createResolvers(resolvers)
+}
+```
+
+To use the newly added root query field in a page query to get the full names of
+all team members, we can write:
+
+```js
+export const query = graphql`
+  {
+    allTeamMembers {
+      ... on Author {
+        fullName
+      }
+      ... on Contributor {
+        fullName
+      }
+    }
+  }
+`
+```
+
+# Extending third-party types
+
+So far, the examples have been dealing with types created from locally available data.
+However, Gatsby also allows to integrate and modify third-party GraphQL schemas.
+
+Usually, those third-party schemas are retrieved from remote sources via introspection
+query with Gatsby's `gatsby-source-graphql` plugin. To customize types integrated from
+a third-party schema, we can use the `createResolvers` API.
+
+## Feeding remote images into `gatsby-image`
+
+As an [example](https://github.com/gatsbyjs/gatsby/blob/master/examples/using-gatsby-source-graphql/gatsby-node.js),
+let's look at how we could use `createResolvers` to feed images from a
+CMS into `gatsby-image` (the assumption is that `gatsby-source-graphql` was configured
+to prefix all types from the third-party schema with `CMS`):
+
+```js:title=gatsby-node.js
+exports.createResolvers = ({
+  actions,
+  cache,
+  createNodeId,
+  createResolvers,
+  store,
+}) => {
+  const { createNode } = actions
+  createResolvers({
+    CMS_Asset: {
+      imageFile: {
+        type: `File`,
+        resolve(source, args, context, info) {
+          return createRemoteFileNode({
+            url: source.url,
+            store,
+            cache,
+            createNode,
+            createNodeId,
+          })
+        },
+      },
+    },
+  })
+}
+```
+
+We create a new `imageFile` field on the `CMS_Asset` type, which will create `File`
+nodes from every value on the `url` field. Since `File` nodes automatically have
+`childImageSharp` convenience fields available, we can feed the images from the CMS
+into `gatsby-image` by simply querying:
+
+```graphql
+query {
+  cms {
+    post {
+      # In this example, the `post.image` field is of type `CMS_Asset`
+      image {
+        # It is important to include all fields in the query which we want to
+        # access in the resolver. In this example we want to make sure to include
+        # the `url` field. In the future, Gatsby might provide a `@projection`
+        # extension to automatically include those fields.
+        url
+        imageFile {
+          childImageSharp {
+            fixed {
+              ...GatsbyImageSharpFixed
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
```

Comment:
I think this page could use a conclusion of some kind since it has so much prose in it. Maybe a quick recap of what was presented in the doc?